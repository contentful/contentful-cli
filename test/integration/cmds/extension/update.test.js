const nixt = require('nixt')
const { resolve } = require('path')

const bin = resolve(__dirname, './../../../../', 'bin')

const configPath = resolve(__dirname, 'fixtures', 'sample-extension.json')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

test('should print help message', done => {
  app()
    .run('extension update --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should exit 1 when no args given', done => {
  app()
    .run('extension update')
    .code(1)
    .expect(result => {
      const regex = /You need to provide a space id./
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})

test('should exit 1 when only space id is given', done => {
  app()
    .run('extension delete --space-id some-id')
    .code(1)
    .expect(result => {
      const regex = /Missing required argument:\s+id/
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})

test('should exit 1 when only id is given', done => {
  app()
    .run('extension update --space-id some-id --id sample-extension')
    .code(1)
    .expect(result => {
      const regex = /Missing required properties:\s+name/
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})

test('should exit 1 when src and srcdoc are omitted', done => {
  app()
    .run(
      'extension update  --space-id some-id --id sample-extension --name foo --field-types Symbol'
    )
    .code(1)
    .expect(result => {
      const regex = /Error: Must contain exactly one of:\s+src, srcdoc/
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})

test('should exit 1 when descriptor given but src and srcdoc still missing', done => {
  app()
    .run(`extension update --space-id some-id --descriptor ${configPath}`)
    .code(1)
    .expect(result => {
      const regex = /Error: Must contain exactly one of:\s+src, srcdoc/
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})
