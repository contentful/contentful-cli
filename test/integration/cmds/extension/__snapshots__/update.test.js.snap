// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message: help data is incorrect 1`] = `
"extension update

Update an extension

Options:
  -h, --help                 Show help                                 [boolean]
  --id                       Extension id                               [string]
  --name                     Extension name                             [string]
  --management-token, --mt   Contentful management API token            [string]
  --space-id                 Space id                                   [string]
  --environment-id           Environment id                             [string]
  --field-types              Field types                                 [array]
  --descriptor               Path to an extension descriptor file       [string]
  --src                      URL to extension bundle                    [string]
  --srcdoc                   Path to extension bundle                   [string]
  --sidebar                  Render the extension in the sidebar       [boolean]
  --version                  Current version of the extension for optimistic
                             locking                                    [number]
  --force                    Force operation without explicit version  [boolean]
  --installation-parameters  JSON string of installation parameter key-value
                             pairs                                      [string]

Copyright 2019 Contentful"
`;
